server/blob/blob-objectstorage/src/main/java/org/apache/james/blob/objectstorage/aws/AwsS3ObjectStorage.java

```diff
@@ -124,7 +128,8 @@ private static TransferManager getTransferManager(AwsS3AuthConfiguration configu
             .standard()
             .withS3Client(amazonS3)
             .withMultipartUploadThreshold(MULTIPART_UPLOAD_THRESHOLD.getValue())
-            .withExecutorFactory(() -> Executors.newFixedThreadPool(MAX_UPLOAD_THREADS))
+            .withExecutorFactory(() -> EXECUTOR_SERVICE)
+            .withShutDownThreadPools(DO_NOT_SHUTDOWN_THREAD_POOL)
```

Comment:
That's better but you still need to shut the threadpool at some time.
You need to add a `@PreDestroy` annotated method in AwsS3ObjectStorage to have guice call it during lifecycle shutdown

Reply:
And as this class only has static methods, you actually need to change it the an instance so that guice can manage it.